src/models/converters/transactions/mod.rs

```diff
@@ -45,72 +45,34 @@ impl MultisigTransaction {
     }
 
     fn transaction_info(&self, info_provider: &mut dyn InfoProvider) -> TransactionInfo {
-        if self.is_settings_change() {
-            // Early exit if it is a setting change
-            return TransactionInfo::SettingsChange(self.to_settings_change());
-        }
-        let token = info_provider.token_info(&self.to).ok();
-        if self.is_erc20_transfer(&token) {
-            TransactionInfo::Transfer(self.to_erc20_transfer(&token.as_ref().unwrap()))
-        } else if self.is_erc721_transfer(&token) {
-            TransactionInfo::Transfer(self.to_erc721_transfer(&token.as_ref().unwrap()))
-        } else if self.is_ether_transfer() {
+        if (self.value_as_uint() > 0 && data_size(&self.data) > 0) || !self.operation.contains(&Operation::CALL) {
+            TransactionInfo::Custom(self.to_custom())
+        } else if self.value_as_uint() > 0 && data_size(&self.data) == 0 {
             TransactionInfo::Transfer(self.to_ether_transfer())
+        } else if self.value_as_uint() == 0 && data_size(&self.data) > 0 && self.safe == self.to {
+            TransactionInfo::SettingsChange(self.to_settings_change())
+        } else if self.data_decoded.as_ref().map(|it| it.is_erc20_transfer_method() || it.is_erc721_transfer_method()).unwrap_or(false) {
+            if !check_sender_or_receiver(&self.data_decoded, &self.safe) {
+                TransactionInfo::Custom(self.to_custom())
+            } else {
+                if let Ok(token) = info_provider.token_info(&self.to) {
+                    match token.token_type {
```

Comment:
Before we also checked, that the token type of the loaded token and the token transfer are the same (this was for some edge cases where weird things happen ... I guess)

Reply:
Right now, that check is happening in the `if` whether we check for erc20 or erc721 transfer methods. I should mix it? Only when we identify the method we proceed to fetch the `tokenInfo` and match the `tokenType`

Reply:
```rust
// ...
} else if self.data_decoded.as_ref().map(|data_decoded|
            data_decoded.is_erc20_transfer_method() || data_decoded.is_erc721_transfer_method()).unwrap_or(false) &&
            check_sender_or_receiver(&self.data_decoded, &self.safe) {
            info_provider.token_info(&self.to)
                .map_or(TransactionInfo::Custom(self.to_custom()),
                        |token|
                            match token.token_type {
                                TokenType::Erc20 => TransactionInfo::Transfer(self.to_erc20_transfer(&token)),
                                TokenType::Erc721 => TransactionInfo::Transfer(self.to_erc721_transfer(&token)),
                                _ => TransactionInfo::Custom(self.to_custom()),
                            },
                )
        } else {
// ...
```

Reply:
nvm ... not sure what I wanted to point at (maybe I was thinking about transfers in ethereum transaction objects). I think this should be fine ... there might be some edge cases, but until liliya finds them I would not optimize for that :P